api-v2/src/plan/entities/plan.entity.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { CourseWarning, IWarning } from '../../../../frontend/src/models/types';
+import { Schedule } from '../../../../common/types';
+import { Student } from '../../student/entities/student.entity';
+
+@Entity()
+export class Plan {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  name: string;
+
+  @ManyToOne(
+    () => Student,
+    student => student.plans,
+    {
+      cascade: true,
+      onDelete: 'CASCADE',
+    },
+  )
+  student: Student;
+
+  @Column({ type: 'json' })
+  schedule: Schedule;
+
+  @Column()
+  major: string;
+
+  @Column()
+  coopCycle: string;
+
+  @Column({ nullable: true })
+  concentration: string;
+
+  @Column({ type: 'smallint', default: 2018 })
+  catalogYear: number;
```

Comment:
why default?

Reply:
To add on to this, i think we should have no default here-it should be an error for the catalog year to not be specified, and i dont think 2018 makes sense as a universal default

Reply:
100%. lowkey i just copy pasta'd the entity types from the previous backend and the previous backend had this for some odd reason. Didn't really use many brain cells when defining the entities haha.

Reply:
LOL npnp